src/rgw/rgw_public_access.cc

```diff
@@ -0,0 +1,39 @@
+#include "rgw_public_access.h"
+#include "rgw_xml.h"
+
+namespace rgw::IAM {
+
+void PublicAccessConfiguration::decode_xml(XMLObj *obj) {
+  RGWXMLDecoder::decode_xml("BlockPublicAcls", BlockPublicAcls, obj);
+  RGWXMLDecoder::decode_xml("IgnorePublicAcls", IgnorePublicAcls, obj);
+  RGWXMLDecoder::decode_xml("BlockPublicPolicy", BlockPublicPolicy, obj);
+  RGWXMLDecoder::decode_xml("RestrictPublicBuckets", RestrictPublicBuckets, obj);
+  //  RGWXMLDecoder::decode_xml("BlockPublicAccessConfiguration", this, obj);
+}
+
+void PublicAccessConfiguration::dump_xml(Formatter *f) const {
+  Formatter::ObjectSection os(*f, "BlockPublicAccessConfiguration");
+  // AWS spec mentions the values to be ALL CAPs, but clients will not
+  // understand this or a mixed case like it is supposed to, hence the need to
+  // manually encode here
+  auto bool_val = [](bool b) -> auto { return b ? "true": "false"; };
```

Comment:
it looks like Formatter::dump_bool() does exactly this

Reply:
thanks will change, encode_xml seemed to have bool with a title case that was breaking s3tests even though it shouldn't :/